evalml/pipelines/utils.py

```diff
@@ -64,33 +64,34 @@ def _get_preprocessing_components(X, y, problem_type, estimator_class, sampler_n
         list[Transformer]: A list of applicable preprocessing components to use with the estimator
     """
 
-    X_pd = X.to_dataframe()
     pp_components = []
-    all_null_cols = X_pd.columns[X_pd.isnull().all()]
+    all_null_cols = X.columns[X.isnull().all()]
     if len(all_null_cols) > 0:
         pp_components.append(DropNullColumns)
-    input_logical_types = set(X.logical_types.values())
-    types_imputer_handles = {logical_types.Boolean, logical_types.Categorical, logical_types.Double, logical_types.Integer}
+    input_logical_types = set(X.ww.logical_types.values())
```

Comment:
I think since this is a private method and is only called in `make_pipeline`, its ok to assume that `infer_feature_types` was called up the stack i.e. in `make_pipeline`. Our test coverage for `make_pipeline` can enforce this.